src/librustdoc/html/render/print_item.rs

```diff
@@ -1936,111 +1938,77 @@ fn document_type_layout<'a, 'cx: 'a>(
     cx: &'a Context<'cx>,
     ty_def_id: DefId,
 ) -> impl fmt::Display + 'a + Captures<'cx> {
-    fn write_size_of_layout(mut w: impl fmt::Write, layout: &LayoutS, tag_size: u64) {
-        if layout.abi.is_unsized() {
-            write!(w, "(unsized)").unwrap();
-        } else {
-            let size = layout.size.bytes() - tag_size;
-            write!(w, "{size} byte{pl}", pl = if size == 1 { "" } else { "s" }).unwrap();
-            if layout.abi.is_uninhabited() {
-                write!(
-                    w,
-                    " (<a href=\"https://doc.rust-lang.org/stable/reference/glossary.html#uninhabited\">uninhabited</a>)"
-                ).unwrap();
-            }
-        }
+    #[derive(Template)]
+    #[template(path = "type_layout.html")]
+    struct TypeLayout<'a, 'cx> {
+        cx: &'a Context<'cx>,
+        ty_def_id: DefId,
```

Comment:
Generally I think we should avoid including `Context`s and `DefId`s directly in `Template`s because that will tend to make us write more Rust code in our templates. For instance, the HTML template below calls `self.variant_name(index.clone())`. It would be better to have a `TypeLayout::new()` that calls the necessary methods on `cx` and `ty_def_id` and stores the results as e.g.

    variants: &'a [(&'str, Size)],